infoschema/infoschema.go

```diff
@@ -403,3 +403,53 @@ func GetBundle(h InfoSchema, ids []int64) *placement.Bundle {
 	}
 	return &placement.Bundle{ID: placement.GroupID(id), Rules: newRules}
 }
+
+// OverrideWithTempTables overrides the old InfoSchema and provide a new InfoSchema.
+func OverrideWithTempTables(is InfoSchema, tables TempTables) InfoSchema {
```

Comment:
InfoSchema with temporary table just merged: https://github.com/pingcap/tidb/pull/25518 . We can use the same schema definition